contracts/modules/CollectionOffers/V1/CollectionOfferBookV1.sol

```diff
@@ -0,0 +1,449 @@
+
+    /// ------------ PRIVATE FUNCTIONS ------------
+
+    /// @notice Checks whether any offers exist for a collection
+    /// @param _collection The address of the ERC-721 collection
+    function _isFirstOffer(address _collection) private view returns (bool) {
+        return (ceilingOfferId[_collection] == 0) && (floorOfferId[_collection] == 0);
+    }
+
+    /// @notice Checks whether a given offer is the only one for a collection
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _offerId The ID of the offer
+    function _isOnlyOffer(address _collection, uint256 _offerId) private view returns (bool) {
+        return (_offerId == floorOfferId[_collection]) && (_offerId == ceilingOfferId[_collection]);
+    }
+
+    /// @notice Checks whether a given offer is the collection ceiling
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _offerId The ID of the offer
+    function _isCeilingOffer(address _collection, uint256 _offerId) private view returns (bool) {
+        return (_offerId == ceilingOfferId[_collection]);
+    }
+
+    /// @notice Checks whether a given offer is the collection floor
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _offerId The ID of the offer
+    function _isFloorOffer(address _collection, uint256 _offerId) private view returns (bool) {
+        return (_offerId == floorOfferId[_collection]);
+    }
+
+    /// @notice Checks whether an offer is greater than the collection ceiling
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _amount The offer amount
+    function _isNewCeiling(address _collection, uint256 _amount) private view returns (bool) {
+        return (_amount > ceilingOfferAmount[_collection]);
+    }
+
+    /// @notice Checks whether an offer is less than or equal to the collection floor
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _amount The offer amount
+    function _isNewFloor(address _collection, uint256 _amount) private view returns (bool) {
+        return (_amount <= floorOfferAmount[_collection]);
+    }
+
+    /// @notice Checks whether an offer can be updated without relocation
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _offerId The ID of the offer
+    /// @param _newAmount The new offer amount
+    /// @param _increase Whether the update is an amount increase or decrease
+    function _isUpdateInPlace(
+        address _collection,
+        uint256 _offerId,
+        uint256 _newAmount,
+        bool _increase
+    ) private view returns (bool) {
+        uint256 nextOffer = offers[_collection][_offerId].nextId;
+        uint256 prevOffer = offers[_collection][_offerId].prevId;
+        return
+            ((_increase == true) && (_newAmount <= offers[_collection][nextOffer].amount)) ||
+            ((_increase == false) && (_newAmount > offers[_collection][prevOffer].amount));
```

Comment:
```solidity
return
    ((_increase == true) && (_newAmount <= offers[_collection][offers[_collection][_offerId].nextId].amount)) ||
    ((_increase == false) && (_newAmount > offers[_collection][offers[_collection][_offerId].prevId].amount));
```

I ran the tests with this code above instead and there should be some gas savings if `_increase` is true, since we don't have to load `prevOffer`.

```
Before:
[PASS] test_DecreaseCeilingInPlace() (gas: 381622)
[PASS] test_DecreaseCeilingToFloor() (gas: 386137)
[PASS] test_DecreaseCeilingToMiddle() (gas: 389288)
[PASS] test_DecreaseFloor() (gas: 384806)
[PASS] test_DecreaseMiddleInPlace() (gas: 381311)
[PASS] test_DecreaseMiddleToFloor() (gas: 386035)
[PASS] test_DecreaseMiddleToMiddle() (gas: 386269)
[PASS] test_IncreaseCeiling() (gas: 384767)
[PASS] test_IncreaseFloorInPlace() (gas: 380793)
[PASS] test_IncreaseFloorToMiddle() (gas: 387346)
[PASS] test_IncreaseFloorToNewCeiling() (gas: 383857)
[PASS] test_IncreaseMiddleInPlace() (gas: 379259)
[PASS] test_IncreaseMiddleToCeiling() (gas: 382625)
[PASS] test_IncreaseMiddleToMiddle() (gas: 385422)
[PASS] test_UpdateFindersFee() (gas: 22932)
--------------------------------------------------------------------------
After:
[PASS] test_DecreaseCeilingInPlace() (gas: 381495)
[PASS] test_DecreaseCeilingToFloor() (gas: 386010)
[PASS] test_DecreaseCeilingToMiddle() (gas: 389161)
[PASS] test_DecreaseFloor() (gas: 384679)
[PASS] test_DecreaseMiddleInPlace() (gas: 381184)
[PASS] test_DecreaseMiddleToFloor() (gas: 385908)
[PASS] test_DecreaseMiddleToMiddle() (gas: 386142)
[PASS] test_IncreaseCeiling() (gas: 384650)
[PASS] test_IncreaseFloorInPlace() (gas: 380676)
[PASS] test_IncreaseFloorToMiddle() (gas: 387229)
[PASS] test_IncreaseFloorToNewCeiling() (gas: 383740)
[PASS] test_IncreaseMiddleInPlace() (gas: 379142)
[PASS] test_IncreaseMiddleToCeiling() (gas: 382508)
[PASS] test_IncreaseMiddleToMiddle() (gas: 385305)
[PASS] test_UpdateFindersFee() (gas: 22932)
```

Reply:
@0xkowloon good looks!! iirc at the time i went with `nextOffer` `prevOffer` for readability, but im down for the optimized version and providing an extra comment or two for clarification if needed.

thanks!